x/incentive/types/msg.go

```diff
@@ -0,0 +1,270 @@
+}
+
+func (msg *MsgBond) GetSigners() []sdk.AccAddress {
+	return checkers.Signers(msg.Account)
+}
+
+// GetSignBytes get the bytes for the message signer to sign on
+func (msg *MsgBond) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func NewMsgBeginUnbonding(account sdk.AccAddress, tier uint32, asset sdk.Coin) *MsgBeginUnbonding {
+	return &MsgBeginUnbonding{
+		Account: account.String(),
+		Tier:    tier,
+		Asset:   asset,
+	}
+}
+
+func (msg *MsgBeginUnbonding) ValidateBasic() error {
+	return validateSenderAssetTier(msg.Account, msg.Tier, &msg.Asset)
+}
+
+func (msg *MsgBeginUnbonding) GetSigners() []sdk.AccAddress {
+	return checkers.Signers(msg.Account)
+}
+
+// GetSignBytes get the bytes for the message signer to sign on
+func (msg *MsgBeginUnbonding) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func NewMsgSponsor(sponsor sdk.AccAddress, programID uint64, asset sdk.Coin) *MsgSponsor {
+	return &MsgSponsor{
+		Sponsor: sponsor.String(),
+		Program: programID,
+		Asset:   asset,
+	}
+}
+
+func (msg *MsgSponsor) ValidateBasic() error {
+	if msg.Program == 0 {
+		return ErrInvalidProgramID.Wrapf("%d", msg.Program)
+	}
+	return validateSenderAsset(msg.Sponsor, &msg.Asset)
+}
+
+func (msg *MsgSponsor) GetSigners() []sdk.AccAddress {
+	return checkers.Signers(msg.Sponsor)
+}
+
+// GetSignBytes get the bytes for the message signer to sign on
+func (msg *MsgSponsor) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func NewMsgGovSetParams(authority, title, description string, params Params) *MsgGovSetParams {
+	return &MsgGovSetParams{
+		Title:       title,
+		Description: description,
+		Params:      params,
+		Authority:   authority,
+	}
+}
+
+func (msg MsgGovSetParams) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(msg.Authority); err != nil {
+		return sdkerrors.Wrap(err, "invalid authority address")
+	}
+	if err := validateProposal(msg.Title, msg.Description); err != nil {
+		return err
+	}
+	return msg.Params.Validate()
+}
+
+// GetSignBytes implements Msg
+func (msg MsgGovSetParams) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(&msg)
+	return sdk.MustSortJSON(bz)
+}
+
+// GetSigners implements Msg
+func (msg MsgGovSetParams) GetSigners() []sdk.AccAddress {
+	return checkers.Signers(msg.Authority)
+}
+
+func NewMsgCreateProgram(authority, title, description string, program IncentiveProgram) *MsgGovCreateProgram {
+	return &MsgGovCreateProgram{
+		Title:       title,
+		Description: description,
+		Program:     program,
+		Authority:   authority,
+	}
+}
+
+func (msg MsgGovCreateProgram) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(msg.Authority); err != nil {
+		return sdkerrors.Wrap(err, "invalid authority address")
+	}
+	if err := validateProposal(msg.Title, msg.Description); err != nil {
```

Comment:
let's add `msg.Authority` to the arguments of `validateProposal`

Reply:
note: will switch from this function to Sai's `util/checkers` implementation